lib/resty/lock.lua

```diff
@@ -100,8 +102,12 @@ function _M.new(_, dict_name, opts)
         exptime = 30
     end
 
-    if timeout and timeout > exptime then
-        timeout = exptime
+    if timeout then
+        timeout = min(timeout, exptime)
+
+        if step then
+            step = min(step, timeout)
+        end
```

Comment:
maybe the following can be simpler?
```
timeout = min(timeout or 5, exptime)
step = min(step or 0.001, timeout)
```

But, we got some logic changed here: `timeout` will always less than `exptime`.
original, `timeout` may large than `exptime`, like:
``` lua
opts = { exptime = 1 }
```
original, we got:
```
exptime = 1
timeout = 5
```
but got we got
```
exptime = 1
timeout = 1
```
Seems the change is acceptable? And we should add some document for this?